### PR TITLE
Fix gcc9 warning in DetectorDescription/DDCMS (missing struct initial…

### PR DESCRIPTION
**Pull request by mrodozov**
#### PR description:
Fixes 
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_amd64_gcc900/CMSSW_11_0_X_2019-10-24-2300/DetectorDescription/DDCMS

#### PR validation:
Compiles without a warning

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: DetectorDescription/DDCMS/interface/Filter.h [hunk 1]
```diff
@@ -28,7 +28,7 @@ namespace cms {
     std::vector<std::string_view> keys;
     std::unique_ptr<Filter> next;
     struct Filter* up;
-    const DDSpecPar* spec;
+    const DDSpecPar* spec = {};
```
Comment (Dr15Jones) [inline DetectorDescription/DDCMS/interface/Filter.h]:
```suggestion
    const DDSpecPar* spec = nullptr;
```
would seem to me to be easier to understand.

**Comment by cvuosalo** [comment]
@ianna: It looks to me like `Filter.spec` is not currently used. Would you agree?

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**